OpenRA.Mods.Common/Projectiles/Bullet.cs

```diff
@@ -297,58 +303,64 @@ bool ShouldExplode(World world)
 				return true;
 
 			// After first bounce, check for targets each tick
-			if (remainingBounces < info.BounceCount && AnyValidTargetsInRadius(world, pos, info.Width, args.SourceActor, true))
+			if (remainingBounces < info.BounceCount && AnyValidTargetsInRadius(world, pos, info.Width, Args.SourceActor, true))
 				return true;
 
 			return false;
 		}
 
-		public IEnumerable<IRenderable> Render(WorldRenderer wr)
+		public virtual IEnumerable<IRenderable> Render(WorldRenderer wr)
 		{
 			if (info.ContrailLength > 0)
 				yield return contrail;
 
-			if (anim == null || ticks >= length)
+			if (ticks >= length)
 				yield break;
 
-			var world = args.SourceActor.World;
+			foreach (var r in RenderAnimation(wr))
```

Comment:
Will likely also have quite a negative peformance overhad with many bullets flying around.

Reply:
I did take inspiration from [Actor.Renderables()](https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Game/Actor.cs#L289), but I can quickly turn it into standard (non-iterator) method.